experimental/module-declarations.md

```diff
@@ -0,0 +1,58 @@
+# [JavaScript Module Declarations][proposal-module-declarations]
+
+## Declarations
+
+### ModuleDeclarations
+
+```js
+interface ModuleDeclaration <: Declaration {
+    type: "ModuleDeclaration";
+    id: Identifier;
+    body: Program;
```

Comment:
Allowing `Program` nodes anywhere other than the root will cause chaos for ESLint rules because that's the assumption right now. Could we maybe created a `ModuleBody` node here instead?

Reply:
For reference this feature is syntactically very similar to TS's namespace/module syntax, and that is exactly what we chose - a block node to enclose the body.

- [TSModuleDeclaration](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/ast-spec/src/declaration/TSModuleDeclaration/spec.ts)
- [TSModuleBlock](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/ast-spec/src/special/TSModuleBlock/spec.ts)